torch/csrc/jit/codegen/cuda/tensor_view.cpp

```diff
@@ -582,17 +555,19 @@ TensorView* TensorView::cache_after() {
     auto this_ca_pos = getThisComputeAtAxis();
     auto this_ca_view = getComputeAtView();
 
-    setComputeAt(consumer, this_ca_pos);
-    consumer->setComputeAt(this_ca_view, rel_ca_pos);
+    setComputeAt(consumer, this_ca_pos, this_ca_pos);
+    consumer->setComputeAt(this_ca_view, this_ca_pos, rel_ca_pos);
   } else {
     // Check users of this TV for computeAt for cache_after on inputs
     for (auto expr : fusion()->unordered_uses(consumer)) {
       for (TensorView* output :
            ir_utils::filterByType<TensorView>(expr->outputs())) {
         if (output->hasComputeAt()) {
-          TransformReplay::replayPasC(consumer, output, -1);
           auto output_ca_pos = output->getThisComputeAtAxis();
-          consumer->setComputeAt(output, output_ca_pos);
+          auto this_pos =
+              TransformReplay::replayPasC(consumer, output, output_ca_pos)
+                  .second;
+          consumer->setComputeAt(output, this_pos, output_ca_pos);
         }
       }
```

Comment:
This part may need more work. See #483. In this PR, I tried to minimize changes in the current behavior.

Reply:
Makes sense.